<?xml version="1.0" ?>
<launch>

  <arg name="use_gui" default="false" />
  <arg name="pub_state" default="1" />
  <arg name="log" default="true" />
  <arg name="offline" default="false" />

  <group if="$(arg offline)">
    <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
    <include file="$(find sis_arm_moveit_config)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>

      <!-- We do not have a robot connected, so publish fake joint states -->
    <node if="$(arg log)" name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="/use_gui" value="$(arg use_gui)"/>
      <rosparam param="/source_list">[/virtual_joint_states]</rosparam>
    </node>

    <!-- Given the published joint states, publish tf for the robot links -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
  </group>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <include file="$(find sis_arm_moveit_config)/launch/moveit_rviz.launch">
    <arg name="config" value="true"/>
  </include>

  <!-- <node pkg="tf" type="static_transform_publisher" name="camear_to_world" args="-0.15 -0.03 0.42 0 0.88 0 base_link camera_link 100" /> -->
  <param name="map_description" command="$(find xacro)/xacro '$(find sis_arm_planning)/map/urdf/factory_map.urdf.xacro'" />
    <node name="map_state_publisher" pkg="robot_state_publisher" type="state_publisher" >
        <remap from="robot_description" to="map_description" />
    </node>


  <node name="pub_frame" pkg="sis_arm_planning" type="pub_frame.py" respawn="false" output="screen" >
  	<param name="/pub_state" value="$(arg pub_state)" />
  </node>


</launch>
